src/indigo/components/SearchBar.jsx

```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import {ButtonToolbar, FormControl} from 'react-bootstrap';
+import Icon from './icons'
+import PropTypes from 'prop-types';
+
+
+class SearchBar extends React.Component {
+  
+  constructor(props) {
+    super(props);
+    this.state = {query: this.props.query};
+    this.changeQuery = this.changeQuery.bind(this)
+    this.submitQuery = this.submitQuery.bind(this)
+  }
+  
+  changeQuery(event) {
+    this.setState({query: event.target.value})
+  }
+  
+  submitQuery(event) {
+    event.preventDefault();
+    this.props.onSubmit(this.state.query);
```

Comment:
myslim, ze `onSubmit` nemusi prijimat query, ak ho bude prijimat `onChange` ... t.j. asi by stacilo `this.props.onSubmit()` ... asi ti to potom lepsie zapasuje aj do kbc-ui

Reply:
jasny, to bylo zbytecny, nevyuzity